static/app/utils/metrics/metricsRequest.tsx

```diff
@@ -122,7 +122,7 @@ class MetricsRequest extends React.Component<Props, State> {
     if (!previousPeriod) {
       return {
         ...commonQuery,
-        statsPeriod,
+        statsPeriod: statsPeriod || undefined,
```

Comment:
@priscilawebdev I'm not sure if this is the correct way to prevent sending an empty string for `?statsPeriod=`.

Either way, with this fix the frontend seems to format the date as 

```
Fri%20Dec%2010%202021%2011%3A18%3A19%20GMT%2B0100%20%28Central%20European%20Standard%20Time%29
```

so I get the response

{"detail":"Fri Dec 10 2021 11:18:19 GMT+0100 (Central European Standard Time) is not a valid ISO8601 date query"}
```

Any idea why the date is formatted that way?

Reply:
sorry...looking into that now 😉

Reply:
that was correct.

I would do something like

```
const period = statsPeriod || undefined;

    const commonQuery = {
      project,
      environment,
      field,
      query: query || undefined,
      groupBy,
      orderBy,
      limit,
      interval: interval ? interval : getInterval({start, end, period}),
      datasource: getMetricsDataSource(),
    };

    if (!previousPeriod) {
      return {
        ...commonQuery,
        statsPeriod: period,
        start: start ?? undefined,
        end: end ?? undefined,
      };
    }

    const doubledStatsPeriod = getPeriod(
      {period, start: undefined, end: undefined},
      {shouldDoublePeriod: true}
    ).statsPeriod;

    return {
      ...commonQuery,
      statsPeriodStart: doubledStatsPeriod || undefined,
      statsPeriodEnd: period ?? DEFAULT_STATS_PERIOD,
    };
```

Reply:
created this frontend PR that solves the issue 

https://github.com/getsentry/sentry/pull/30717